share/functions/__fish_complete_zfs_write_once_properties.fish

```diff
@@ -0,0 +1,33 @@
+function __fish_complete_zfs_write_once_properties -d "Completes with ZFS properties which can only be written at filesystem creation, and only be read thereafter"
+	set -l OS ""
+	if passwd --help >/dev/null ^&1
+		set OS "Linux"
+	else # Not Linux, so use the ugly uname
+		set -l os_type (uname)
+		switch $os_type
+			case Darwin
+				set OS "macOS"
+			case FreeBSD
+				set OS "FreeBSD"
+			case SunOS
+				set OS "SunOS"
+			# Others?
+			case "*"
+				set OS "unknown"
+		end
+	end
+	echo -e "normalization\t"(_ "Unicode normalization")" (none, formC, formD, formKC, formKD)"
+	echo -e "utf8only\t"(_ "Reject non-UTF-8-compliant filenames")" (on, off)"
+	if test $OS = "Linux"
+		echo -e "overlay\t"(_ "Allow overlay mount")" (on, off)"
+		if command -s sestatus >/dev/null ^&1 # SELinux is enabled
```

Comment:
It seems that `share/completions/mkdir.fish` could use that too.

Reply:
Yes. A bunch could - it's supported since 2.5.0, and we haven't gotten around to modifying all the scripts yet. It's just cleanup, so it's not too high priority.